src/pages/studyView/StudyViewPageStore.ts

```diff
@@ -2512,23 +2512,31 @@ export class StudyViewPageStore {
     private loadSettings(chartSettngs: ChartUserSetting[]){
         this.clearPageSettings();
         _.map(chartSettngs, chartUserSettings => {
+            const regex = /^(?:PATIENT_|SAMPLE_)([a-zA-Z0-9_]+)/i;
```

Comment:
Let's move this to study  view util.  Also the similar logic converting the chart id to clinical attribute should be put together(getClinicalDataBySamples method)

Reply:
I don't see any similarity between getClinicalDataBySamples and chart id. Do you mean to move getClinicalDataBySamples method to utils?